### ISSUE TITLE
softmax_cross_entropy documentation not consistent with its behavior

### ISSUE THREAD

**Issue by koreyou** [chainer/chainer#5918]
# Problem description

The documentation for `softmax_cross_entropy` states that return value is:

```
A variable holding a scalar array of the cross entropy loss. If reduce is 'mean', it is a scalar array. If reduce is 'no', the shape is same as that of x.
```

https://docs.chainer.org/en/v5.0.0/reference/generated/chainer.functions.softmax_cross_entropy.html

But current implementation behaves like following:

```
x = np.random.random((2, 3)).astype(np.float32)
chainer.functions.softmax_cross_entropy(
    x,
    np.array([1, 0], np.int32),
    reduce='no'
)
```

will give 2D array `variable([1.0142076, 1.3368266])` of shape `(2, )`.

This might be related to #2182 .

## Environment to reproduce

```
Platform: Linux-4.14.65+-x86_64-with-Ubuntu-18.04-bionic
Chainer: 5.0.0
NumPy: 1.14.6
CuPy:
  CuPy Version          : 5.0.0
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 9020
  CUDA Driver Version   : 9020
  CUDA Runtime Version  : 9020
  cuDNN Build Version   : 7201
  cuDNN Version         : 7201
  NCCL Build Version    : 2213
iDeep: 2.0.0.post3
```

# Suggestion

Easy fix would be to change documentation to be consistent with current behavior. *But*, I belive the current behaviour is wrong.

First reason is that, `sigmoid_cross_entropy` returns array of same shape as `x` when `reduce='no'`is passed. Behaviours for the similar functions should be similar.

Second reason is that returning tensor of same shape as `x` is preferred for better flexibility. For example, when we want to implement focal loss ( https://arxiv.org/abs/1708.02002 ), we need tensor before the reduction.

**Comment by toslunar** [comment]
I'll fix the document.

- I don't call "(p log q) before summation" a cross entropy.  It could be provided as a new function with a different name.
  - Could you directly use `F.log_softmax` so far?
- `softmax_cross_entropy` and `sigmoid_cross_entropy` interprets `reduce='no'` option consistently in the sense that both of them return an array of the same shape as `t`.

**Comment by toslunar** [comment]
The documentation is fixed by #5965.  Feel free to open a new issue if you'd like to change the behavior.
